patches/api/0495-Add-new-TickRateChangedEvent.patch

```diff
@@ -0,0 +1,82 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Axionize <154778082+Axionize@users.noreply.github.com>
+Date: Tue, 8 Oct 2024 20:38:01 -0400
+Subject: [PATCH] Add new TickRateChangedEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/server/TickRateChangeEvent.java b/src/main/java/org/bukkit/event/server/TickRateChangeEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..5a72aab79fdde77f6ac25afbd586f03bf8db732d
+--- /dev/null
++++ b/src/main/java/org/bukkit/event/server/TickRateChangeEvent.java
+@@ -0,0 +1,70 @@
++package org.bukkit.event.server;
++
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.ApiStatus;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++import org.jspecify.annotations.NullMarked;
++
++/**
++ * Fired when the server's target tickrate changes
++ */
++@NullMarked
++public class TickRateChangeEvent extends ServerEvent implements Cancellable {
++
++    private static final HandlerList HANDLER_LIST = new HandlerList();
++
++    private final CommandSender commandSender;
++    private float tickRate;
++    private boolean cancelled;
++
++    @ApiStatus.Internal
++    public TickRateChangeEvent(CommandSender commandSender, float tickRate) {
++        this.commandSender = commandSender;
++        this.tickRate = tickRate;
++    }
++
++    /**
++     * @return {@link org.bukkit.command.CommandSender} representing source of the /tick rate command
++     * Will return null if the tick rate was set by a plugin or setTickRate() was otherwise directly called
++     */
++    @Nullable
++    public CommandSender getCommandSender() {
```

Comment:
Brought up during some chatting regarding this PR, but it would probably be worth exposing the entire io.papermc.paper.command.brigadier.CommandSourceStack. Exposes more context over what is essentially a "Someone ran the tick command" event.

Reply:
Done

Was the chatting about this PR in a public channel I can see?

Reply:
Thanks for the quick fix, no that was just a fleeting discussion in internal channels.
I'll try to make time for merging it this weekend, which usually happens in one of our discord VCs.